### ISSUE TITLE
Disallow definition of operators covered by type classes

### ISSUE THREAD

**Issue by mlutze** [flix/flix#3247]
```scala
def |||(t1: Ty, t2: Ty): Ty = Or(t1, t2)
...
def anyTag(): Ty = Not(anyProd() ||| anyArrow() ||| anyInt())
```
gives
```
>> No instance of the 'BitwiseOr' class for the type 'Ty'.

39 | def anyTag(): Ty = Not(anyProd() ||| anyArrow() ||| anyInt())
                                      ^^^
                                      missing instance
```

But the issue is really at the definition site, as that `|||` op can never be used.

**Comment by magnus-madsen** [comment]
Because its reserved right?

**Comment by mlutze** [comment]
Yes, kind of.

The issue is that `|||` is desugared to `BitwiseOr.bitwiseOr`. So even though we have a local definition of `|||`, it's ignored.

**Comment by magnus-madsen** [comment]
OK

**Comment by mlutze** [comment]
> `Bool.flix` defines `not`, `and`, and `or`, which will be disallowed by this change.
> 
> Should we be adding all keywords to the list? Including non-operators like `pub`, `def`, `class`, etc.

**Comment by magnus-madsen** [comment]
I think so. I think agreed in another ticket to ban all keywords etc.

**Comment by mlutze** [comment]
Blocked by https://github.com/flix/flix/issues/3307

**Comment by magnus-madsen** [comment]
This works:

```
enum Ty {
    case F
}

def |||(t1: Ty, t2: Ty): Ty = ???

def anyTag(): Ty = Ty.F ||| Ty.F
```

is that not good enough? `not` is not a keyword.
